### PR TITLE
9 per participant results

### PR DESCRIPTION
**Pull request by wasade**
depends on #167, #168, and #169

### CODE REVIEW

**Comment by wasade** [comment]
bam bam, passing!

File: americangut/per_sample.py [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    sample_type : str, {fecal, oral, skin}
+        The sample type.
+
+    chp_path : str
+        The path to the current processing chapter.
+
+    gradient_color_by : str
+        The taxon to color by for the gradient PCoA.
+
+    barchart_categories : iterable
```
Comment (mortonjt) [inline americangut/per_sample.py]: iterable of strings?

File: americangut/per_sample.py [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+
+    return id_.split('.')[1]
+
+
+def partition_samples_by_bodysite(md, site_to_function):
+    """Yield the processing function and associated sample IDs
+
+    Parameters
+    ----------
+    md : DataFrame
```
Comment (mortonjt) [inline americangut/per_sample.py]: pd.DataFrame?

File: americangut/per_sample.py [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+        The metadata to examine. It is expected that this DataFrame contain a
+        column named SIMPLE_BODY_SITE.
+    site_to_function : list of tuple
+        A mapping of a body site category value to a processing function.
+
+    Returns
+    -------
+    generator
+        (fucntion, iterable)
```
Comment (mortonjt) [inline americangut/per_sample.py]: function

File: americangut/per_sample.py [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+        specific details (e.g., taxon to use for gradient coloring).  The
+        return from a function is expected to be a dict in which every input
+        sample ID from the passed in list is represented as a key, and the
+        value is a list of str where each str indicates an error occured -- an
+        empty list would indicate no observed errors for the sample.
+
+    opts : dict
+        Paths and relevant processing options.
+
+    ids : iterable
```
Comment (mortonjt) [inline americangut/per_sample.py]: Same here

File: americangut/per_sample.py [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+    """Iteratively execute a system call over sample IDs
+
+    Parameters
+    ----------
+    cmd_fmt : str
+        The format of the command to execute. It is expected that there
+        is a single string format to be done and it should take a sample
+        ID
+    sample_ids : iterable
+        A list of sample IDs of interest
```
Comment (mortonjt) [inline americangut/per_sample.py]: Same here

File: americangut/per_sample.py [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+    opts : dict
+        A dict of relevant opts.
+
+    sample_ids : iterable
+        A list of sample IDs of interest
+
+    Returns
+    -------
+    dict
+        A dict containing each sample ID and any errors observed or None if
```
Comment (mortonjt) [inline americangut/per_sample.py]: Keys are sampleIDs, and values are errors correct?

File: ipynb/primary-processing/9-per_participant_results.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+>>> process_fecal = partial(agps.sample_type_processor, fecal_functions, fecal_opts)
+>>> process_oral = partial(agps.sample_type_processor, oral_functions, oral_opts)
+>>> process_skin = partial(agps.sample_type_processor, skin_functions, skin_opts)
+```
+
+And before the fun starts, let's stage static aspects of the participant results. These are things like the American Gut logo, the result template, etc.
+
+```python
+>>> agru.stage_static_files('fecal', chp_path)
+>>> agru.stage_static_files('oralskin', chp_path)
```
Comment (mortonjt) [inline ipynb/primary-processing/9-per_participant_results.md]: Are you sure you want `oralskin` together?

Comment (wasade) [inline ipynb/primary-processing/9-per_participant_results.md]: They're the same template at least right now. That's still legacy code...

**Comment by mortonjt** [comment]
Very minor comments.  Otherwise I think it looks good.

File: americangut/per_sample.py (hunk 1 above)
Comment (wasade) [inline americangut/per_sample.py]: done, and modified all relevant instances of this in the documentation

File: americangut/per_sample.py (hunk 2 above)
Comment (wasade) [inline americangut/per_sample.py]: done

File: americangut/per_sample.py (hunk 3 above)
Comment (wasade) [inline americangut/per_sample.py]: done

**Pull request merged**